### PR TITLE
SPOI-10007 #resolve : Improved performance of table when scrolling

### PR DESCRIPTION
**Pull request by wvary**
1. Improved performance by changing from setTimeout to requestAnimationFrame method
    - Browser does not guaranty the scroll code to execute within 100 millis
    - We've seen cases where scroll code doesn't run after 2 seconds
2. Move the header table into the div to ensure correct alignment when the scrollbar is shown
3. Implemented caching to prevent running filtering and sorting again if already done once
    - Cache does get clear if the data is udpated
4. Changed highlight row feature to run the function during binding
    - This approach prevents changing the actual data
5. Change version to 2.0.0
    
Possible future improvements:
 i. Better handling of different row heights
ii. Throttle the scrolling code (maybe by cancelling the requestAnimationFrame)

### CODE REVIEW

File: src/directives/mlhrTable.js [hunk 1]
```diff
@@ -188,7 +187,34 @@ angular.module('datatorrent.mlhrTable.directives.mlhrTable', [
...
     var scrollDeferred;
-    var debouncedScrollHandler = debounce(function() {
+    var scrollTopSaved = -1;
+
+    // determine requestAnimationFrame compabitility
+    var raf = window.requestAnimationFrame
+            || window.mozRequestAnimationFrame
+            || window.webkitRequestAnimationFrame
+            || window.msRequestAnimationFrame
+            || function(f) { return setTimeout(f, 100) };
```
Comment (peyao) [inline src/directives/mlhrTable.js]: Why does this setTimeout have a 100 ms delay?

Comment (wvary) [inline src/directives/mlhrTable.js]: @peyao Good catch, I'll update it to use the debounce option value.  It's a fallback function anyway, but will use it.

I didn't mention in the PR, but I can to update the documentation and unit tests for this change, but I want to get the acceptance of feature and code change before I do that.

File: app/scripts/controllers/main.js [hunk 2]
```diff
@@ -109,8 +109,9 @@ angular.module('datatorrent.mlhrTable.ghPage')
     // Generates `num` random rows
     function genRows(num){
       var retVal = [];
+      var id = ($scope.my_table_data[0] ? $scope.my_table_data[0].id + 1 : 0);
       for (var i=0; i < num; i++) {
-        retVal.push(genRow(i));
+        retVal.push(genRow(id++));
```
Comment (sashadt) [inline app/scripts/controllers/main.js]: ID's seem to be duplicated - maybe async issue?

File: src/directives/mlhrTable.js (hunk 1 above)
Comment (wvary) [inline src/directives/mlhrTable.js]: @peyao Changed code to use scope.options.scrollDebounce.

File: app/scripts/controllers/main.js (hunk 2 above)
Comment (wvary) [inline app/scripts/controllers/main.js]: @sashadt Fixed duplicate id issue.

**Pull request merged**